### PR TITLE
samples: nrf9160: AT monitor

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add monitor initialization.
Add modem shutdown.
Add explanation when network is not connecting.

### CODE REVIEW

File: samples/nrf9160/at_monitor/src/main.c [hunk 1]
```diff
@@ -152,12 +147,17 @@ void main(void)
...
 
 	if (cereg_status == HOME) {
 		printk("Network connection ready\n");
 	} else {
-		printk("No network connection\n");
+		if (err == -EAGAIN) {
+			printk("Network connection timed out\n");
+		} else {
+			printk("No network connection found\n");
+		}
```
Comment (lemrey) [inline samples/nrf9160/at_monitor/src/main.c:159]: I would omit this `else` clause, I am not sure when we'd need that.

**Review by lemrey — COMMENTED** [review]
Looks good, left one small comment.
Also I would rephrase the git commit title slightly, so we prefix the sample name e.g.: `samples: nrf9160: at_monitor: fix..`

**Review by lemrey — APPROVED** [review]

**Comment by lemrey** [comment]
@umapraseeda could you please have a look at the doc changes?

**Review by umapraseeda — APPROVED** [review]
